public/scripts/utils.js

```diff
@@ -391,6 +391,26 @@ export function getStringHash(str, seed = 0) {
     return 4294967296 * (2097151 & h2) + (h1 >>> 0);
 }
 
+/**
+ * Copy text to clipboard. Use navigator.clipboard.writeText if available, otherwise use document.execCommand.
+ * @param {string} text - The text to copy to the clipboard.
+ * @returns {Promise<void>} A promise that resolves when the text has been copied to the clipboard.
+ */
+export function copyText(text) {
+    if (navigator.clipboard) {
+        return navigator.clipboard.writeText(text);
+    }
+
+    const parent = document.querySelector('dialog[open]:last-of-type') ?? document.body;
+    const textArea = document.createElement('textarea');
```

Comment:
Sneaky, clever.
Doesn't this lead to a possible visible draw/redraw of the UI with the textarea appearing?
Maybe make adding a CSS class to make it absolute and out of screen would make 100% sure.

Reply:
Okay, good then.